### PR TITLE
Fibonacci

### PR DESCRIPTION
**Pull request by Martyna-Tu**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @Martyna-Tu

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Martyna-Tu for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,27 @@
 #pragma once
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
-}
+    if (sequence == 0 ) {
+	    return 0;
+    }else if (sequence == 1) {
+	    return 1;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:7]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+    }else {
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:8]: else is redundant

File: homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
-}
+    if (sequence == 0 ) {
+	    return 0;
+    }else if (sequence == 1) {
+	    return 1;
+    }else {    
+    	int x = 0, y = 1, z = 0;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:9]: One definition/ declaration per line.

File: homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -1,11 +1,27 @@
...
+ }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence == 0 ) {
+        return 0;
+    }else if(sequence == 1){
+        return 1;
+    }else {
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:24]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 5]
```diff
@@ -1,11 +1,27 @@
...
+        return fibonacci_recursive(sequence-1)+fibonacci_recursive(sequence-2);
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:25]: Add space between operations like '+' '-' etc...

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**